src/main/ts/ui/Dialog.ts

```diff
@@ -0,0 +1,137 @@
+import { Arr } from "@ephox/katamari";
+import * as Utils from '../core/Utils';
+import { Editor } from "tinymce";
+import { PublicDialog } from '@ephox/bridge';
+import { PredicateFilter, SugarElement, Traverse } from "@ephox/sugar";
+import { EnhancedListDialogData } from "./DialogTypes";
+
+const styleOptions: string[] = ["circle", "disc", "square", "lower-alpha", "lower-greek", "lower-roman", "upper-alpha", "upper-roman"];
+
+const getStyleOptions = (options: string[]) => {
+  return Arr.map(options, (o) => ({
+    text: o,
+    value: o
+  }));
+};
+
+const items: PublicDialog.BodyComponentSpec[] = [
+  {
+    type: 'listbox',
+    name: 'listStyle',
+    label: 'List Style',
+    items: getStyleOptions(styleOptions)
+  },
+  {
+    type: 'input',
+    name: 'paddingLeft',
+    label: 'Padding Left',
+    inputMode: 'numeric'
+  },
+];
+
+const getPanelItems = (selElement: SugarElement<Node>): PublicDialog.BodyComponentSpec[] => {
+  if (Utils.isNestedList(selElement)) {
+    return items.concat({
+      type: 'listbox',
```

Comment:
Because this item doesn't require any information from `selElement`, you could also put it outside the function, and then this just becomes a straight ternary. Who doesn't love ternaries....

```
return [
  ...items,
  Utils.isNestedList(selElement) ? [ applyListbox ] : [ ]
];
```